test/sqllogictest/transform/lifting.slt

```diff
@@ -637,40 +637,51 @@ EOF
 query T multiline
 explain select min(a) from t_pk where a between 38 and 195 and a = (select a from t where a = 1308);
 ----
-%0 =
+%0 = Let l0 =
+| Get materialize.public.t (u1)
+| Filter (#0 = 1308)
+
+%1 =
 | Get materialize.public.t_pk (u3)
 | ArrangeBy (#0)
 
-%1 =
-| Get materialize.public.t (u1)
-| Filter (#0 = 1308)
+%2 =
+| Get %0 (l0)
+| Project (#0)
+
+%3 =
+| Get %0 (l0)
 | Reduce group=()
 | | agg count(true)
 | Filter (err: more than one record produced in subquery), (#0 > 1)
 | Map (err: more than one record produced in subquery)
+| Project (#1)
 
-%2 = Let l0 =
-| Join %0 %1 (= #0 #3)
-| | implementation = Differential %1 %0.(#0)
+%4 =
+| Union %2 %3
+
+%5 = Let l1 =
+| Join %1 %4 (= #0 #2)
+| | implementation = Differential %4 %1.(#0)
 | | demand = (#0)
 | Filter (#0 <= 195), (#0 >= 38)
```

Comment:
These filters are originally around `t_pk`, and the information from them cannot get propagated to `t` until they get lifted after `JoinImplementation` runs. 

#3044 should allow this information to be propagated to `t` before the JoinImplementation stage.

Reply:
Sweet; I was unclear on what was going on here; thank you!

Reply:
Oh, I wasn't aware of #3044. That seems to have some overlap with the new `PredicatePullup` transform in #7471.